include/internal/list.h

```diff
@@ -12,6 +12,15 @@
 # pragma once
 
 # include <string.h>
+# include <assert.h>
+
+# ifdef NDEBUG
+#  define OSSL_LIST_CHK(x)
```

Comment:
OSSL_LIST_CHK seems rather pointless. Why not just use assert directly since it isn't compiled in non-debug builds anyway?

Reply:
It's not quite the same, this preprocesses to nothing rather than an empty statement.
Why I did this, I don't remember.